sycl/include/sycl/reduction.hpp

```diff
@@ -2328,9 +2328,7 @@ template <
     typename = std::enable_if_t<!has_known_identity<BinaryOperation, T>::value>>
 detail::reduction_impl<
     T, BinaryOperation, 0, 1,
-    accessor<T, 1, access::mode::read_write, access::target::device,
-             access::placeholder::true_t,
-             ext::oneapi::accessor_property_list<>>>
+    accessor<T, 1, access::mode::read_write, access::target::device>>
```

Comment:
I'd prefer to change the exception below to a template param-dependent static assert.

Reply:
To help testing in the SYCL-CTS I would prefer a runtime exception so we can at least compile and run the subset that we know works while this is being implemented.

Reply:
But the real customers would prefer the static error, not runtime one.

Reply:
@steffenlarsen do you mean an exception for now and soon a static_assert once the SYCL CTS is complete?
Otherwise there are a lot of macros in the CTS today already to skip failing tests and it does not looks appealing to lower the SYCL runtime quality just for the CTS.
@bader thoughts?

Reply:
The intention is not to leave this function unimplemented for much longer and I worry that the CTS tests for reductions aren't currently in a state where the test triggering this specific case is easy to disable without disabling larger parts of the tests. However, if I am wrong I agree that the `static_assert` is the better user experience, but seeing as the CTS is the first to find the obscure compilation error (that we know of) I am not too worried that users will experience this much ahead of it being implemented.

Reply:
Merging this PR to unblock testing. If we figure there is a good way to work around the failure while having a `static_assert` it is a simple change.